lib/fizzy/types.hpp

```diff
@@ -29,7 +29,7 @@ struct FuncType
 struct Limits
 {
     uint32_t min;
-    uint32_t max;
+    std::optional<uint32_t> max;
```

Comment:
This field is used twice in the spec, for `memory` and `tables`. Therefore only used during instantiation and the overhead of optional is acceptable there.